subprojects/java-compiler-plugin/src/test/groovy/com/gradle/internal/compiler/java/listeners/ConstantsCollectorTest.groovy

```diff
@@ -42,11 +42,12 @@ class ConstantsCollectorTest extends AbstractCompilerPluginTest {
         )
     }
 
-    def "should always return self as a private constant origin class"() {
+    def "should not return self as a constant origin class"() {
```

Comment:
Sure. So adding self as constant origin is not needed since if class is changed it is compiled and it does not need self reference. 

But in the implementation of constant tracking we tracked visited classes that way, since that info was used to merge previous and new constant tracking result. That was later simplified in: https://github.com/gradle/gradle/commit/762c7d1e4f88f89fae0ac27119de381fcbf9ecf3 (check commit message under _Simplify ConstantDependentsMapping_): self tracking is not used anymore and changed classes info is used when merging results. But the code in Compiler plugin was not removed. 

Additionally you can also see in:
https://github.com/gradle/gradle/blob/8173acdb88b1fdaf7a7cda9e583800628ad145fb/subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/compilerapi/constants/ConstantsAnalysisResult.java#L34-L44
that when collecting constant dependents there is a check that constant dependent is not equal to constant origin. These two methods are actually called when collecting dependents in compiler plugin. So actually adding self as constant origin already does nothing.

I hope that makes sense.